agent/providers/lib/v8.go

```diff
@@ -97,7 +97,7 @@ func (v *V8) Deploy() error {
 	subject := fmt.Sprintf("agentint.%s.trigger", v.vmID)
 	_, err := v.nc.Subscribe(subject, func(msg *nats.Msg) {
 		startTime := time.Now()
-		val, err := v.Execute(msg.Header.Get(agentapi.NexTriggerSubject), msg.Data)
+		val, err := v.Execute(msg.Header, msg.Data)
```

Comment:
produce a context here. First, create a context by extracting the otel header from `msg.Header`, which is done via https://pkg.go.dev/go.opentelemetry.io/otel/propagation#TraceContext.Extract 

Calling extract on the nats message header will produce a `context.Context` . None of our code should ever manually copy a traceparent into an explicitly named header variable.

Reply:
so:
1. create a context via otel extract from the nats header
1. create a new context from the previous by adding the trigger subject value to it